src/uri/path.rs

```diff
@@ -504,4 +509,32 @@ mod tests {
         assert!(path_and_query > "/a/world&foo=bar".to_string());
         assert!("/a/world&foo=bar".to_string() < path_and_query);
     }
+
+    #[test]
+    fn double_percent_path() {
+        let double_percent_path = "/your.js?bn=%%val";
+
+        assert!(double_percent_path.parse::<PathAndQuery>().is_ok());
+
+        let path: PathAndQuery = double_percent_path.parse().unwrap();
+        assert_eq!(path, double_percent_path);
+
+        let double_percent_path = "/path%%";
```

Comment:
I have added few additional test cases now if path ends with double percent no error is returned on the other hand if ends with 1 percent `ErrorKind::InvalidUriChar` error is returned